rixs/process2d.py

```diff
@@ -0,0 +1,550 @@
+    pointsize : float
+        multiply all point sizes by a fixed factor
+    **kwargs : dictionary
+        passed onto matplotlib.pyplot.scatter.
+    Returns
+    ----------
+    image_artist : matplotlib artist object
+        artist from image scatter plot
+    """
+    ax.cla()
+    ax.set_facecolor('black')
+    defaults = {'c': 'white', 'edgecolors': 'white', 'alpha': 0.5,
+                'pointsize': pointsize}
+
+    kwargs.update({key: val for key, val in defaults.items()
+                   if key not in kwargs})
+
+    pointsize = kwargs.pop('pointsize')
+    image_artist = ax.scatter(photon_events[:, 0], photon_events[:, 1],
+                              s=photon_events[:, 2]*pointsize, **kwargs)
+    return image_artist
+
+
+def plot_pcolor(ax, photon_events, cax=None, bins=None, **kwargs):
+    """Make an pcolorfast image of the photon_events
+    This is effcient when there are a large RIXS intensity
+
+    Parameters
+    ----------
+    ax : matplotlib axes object
+        axes for plotting on
+    photon_events : array
+        two column x, y, Iph photon locations and intensities
+    cax : matplotlib axes object
+        use to plot colorbar to a specific axis
+    bins : int or array_like or [int, int] or [array, array]
+        The bin specification in y then x order:
+            * None chooses bin edges from int/2, int/2 +1 etc.
+              with partially populated edges removed.
+            * If int, the number of bins for the two dimensions (ny=nx=bins).
+            * If array_like, the bin edges for the two dimensions
+              (y_edges=x_edges=bins).
+            * If [int, int], the number of bins in each dimension
+              (ny, nx = bins).
+            * If [array, array], the bin edges in each dimension
+              (y_edges, x_edges = bins).
+            * A combination [int, array] or [array, int], where int
+              is the number of bins and array is the bin edges.
+
+    **kwargs : dictionary
+        passed onto matplotlib.pyplot.imshow.
+    Returns
+    ----------
+    image_artist : matplotlib artist object
+        artist from image plot
+    cb_artist : matplotlib artist object
+        artist from colorbar associated with plot
+    """
+    ax.cla()
+
+    image_output = photon_events_to_image(photon_events, bins=bins)
+    x_centers, y_centers, image, x_edges, y_edges = image_output
+
+    defaults = {'cmap': 'gray',
+                'vmin': np.nanpercentile(image, 1),
+                'vmax': np.nanpercentile(photon_events[:, 2], 99)}
+
+    kwargs.update({key: val for key, val in defaults.items()
+                   if key not in kwargs})
+
+    xlim = (x_edges.min(), x_edges.max())
+    ylim = (y_edges.min(), y_edges.max())
+    image_artist = ax.pcolorfast(xlim, ylim, image, **kwargs)
+    cb_artist = plt.colorbar(image_artist, ax=ax, cax=cax)
+    ax.axis('tight')
+
+    return image_artist, cb_artist
+
+
+def poly(x, x2=0., x1=0., x0=500.):
```

Comment:
This should be replaced with numpys builtin polynomial function (https://docs.scipy.org/doc/numpy-1.13.0/reference/routines.polynomials.polynomial.html)

Reply:
I did consider this. 

The problem I have is that lmfit cannot take np.polyval as an input as lmfit cannot handle the arbitrary number of *args in np.polyval.

So I don't know how to do this unless I wrap np.poly in another function. 

Note that  np.polyval looks as below under the hood -- so essentially like process2d.poly except less readable. 

    p = NX.asarray(p)
    if isinstance(x, poly1d):
        y = 0
    else:
        x = NX.asarray(x)
        y = NX.zeros_like(x)
    for i in range(len(p)):
        y = x * y + p[i]
    return y

Reply:
I would suggest using numpy.polyfit for fitting polynominals. (https://docs.scipy.org/doc/numpy/reference/generated/numpy.polyfit.html).

lmfit even specify that it is better to use numpy.polyfit for pure polynomial fitting and that they are included in lmfit only for composite model fitting.

>Linear and Polynomial Models
>These models correspond to polynomials of some degree. Of course, lmfit is a very inefficient way to do linear regression (see numpy.polyfit or scipy.stats.linregress), but these models may be useful as one of many components of a composite model. 

quote from https://lmfit.github.io/lmfit-py/builtin_models.html#linear-and-polynomial-models

Reply:
Yes, I think it's reasonable to change to numpy.polyfit .